proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,113 @@
+<!-- {% raw %} -->
+
+# Support for in-process, Out-of-process, or both for the c style compiler api
+
+* Proposal: [NNNN](INF-NNNN-inproc-outofproc-compiler-api-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style api will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+This document is to propose building only the out-of-process support which
+gives the caller a more performant and secure solution.
+
+Visual Studio uses an out of process architecture for compilation which helps
+better utilize the power of the machine performing the compilation as well as
+isolating the code editor from any crashes that may occur during compilation.
```

Comment:
I don't quite get what Visual Studio has to do with anything here?

Reply:
I was using it as an example.  It compiling things using out of process compilation.  I could remove this if you think it is too confusion.

Reply:
It adds confusion.  Pretty much _anything_ that compiles by launching the compiler as a separate .exe is doing out of proc compilation.

Reply:
ha!  I didn't think about that!  Good point!

Reply:
fixed